clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/NativeMetadataRepositoryViewAdapter.java

```diff
@@ -0,0 +1,234 @@
+    if (store == null) {
+      return null;
+    }
+    if (!VeniceView.isViewStore(storeName)) {
+      return store;
+    }
+    // It's a view store, so we need to create a ReadOnlyViewStore
+    return new ReadOnlyViewStore(store, storeName);
+  }
+
+  @Override
+  public Store getStoreOrThrow(String storeName) throws VeniceNoStoreException {
+    Store store = getStore(storeName);
+    if (store != null) {
+      return store;
+    }
+    throw new VeniceNoStoreException(storeName);
+  }
+
+  @Override
+  public boolean hasStore(String storeName) {
+    return nativeMetadataRepository.hasStore(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public Store refreshOneStore(String storeName) {
+    return nativeMetadataRepository.refreshOneStore(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public List<Store> getAllStores() {
+    return nativeMetadataRepository.getAllStores();
+  }
+
+  @Override
+  public long getTotalStoreReadQuota() {
+    return nativeMetadataRepository.getTotalStoreReadQuota();
+  }
+
+  @Override
+  public void registerStoreDataChangedListener(StoreDataChangedListener listener) {
+    StoreDataChangedListener adapterListener = new StoreDataChangedListenerViewAdapter(listener, this);
+    nativeMetadataRepository.registerStoreDataChangedListener(adapterListener);
+    storeDataChangedAdapterMap.put(listener, adapterListener);
+  }
+
+  @Override
+  public void unregisterStoreDataChangedListener(StoreDataChangedListener listener) {
+    storeDataChangedAdapterMap.compute(listener, (dataChangeListener, dataChangedListenerAdapter) -> {
+      if (dataChangedListenerAdapter != null) {
+        nativeMetadataRepository.unregisterStoreDataChangedListener(dataChangedListenerAdapter);
+      }
+      return dataChangedListenerAdapter;
+    });
+  }
+
+  @Override
+  public int getBatchGetLimit(String storeName) {
+    return nativeMetadataRepository.getBatchGetLimit(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public boolean isReadComputationEnabled(String storeName) {
+    return nativeMetadataRepository.isReadComputationEnabled(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public void subscribe(String storeName) throws InterruptedException {
+    String internalStoreName = VeniceView.getStoreName(storeName);
+    nativeMetadataRepository.subscribe(internalStoreName);
+    subscribedStoreMap.compute(internalStoreName, (k, s) -> {
+      Set<String> stores = s == null ? new ConcurrentSkipListSet<>() : s;
+      stores.add(storeName);
+      return stores;
+    });
+  }
+
+  @Override
+  public void unsubscribe(String storeName) {
+    String internalStoreName = VeniceView.getStoreName(storeName);
+    subscribedStoreMap.compute(internalStoreName, (k, s) -> {
+      if (s != null) {
+        s.remove(storeName);
+        if (s.isEmpty()) {
+          nativeMetadataRepository.unsubscribe(internalStoreName);
```

Comment:
Shall we return `null` to remove the entry when the set becomes empty?
Again, please verify whether returning `null` can remove the entry from map or not.